File: 0011.ContainerWithMostWater.go
func maxArea(height []int) int {
    ans := 0
    left, right := 0, len(height)-1
    for left < right {
        if height[left] <= height[right] {
            ans = max(ans, (right-left) * height[left])
            left++
        } else {
            ans = max(ans, (right-left) * height[right])
            right--
        }
    }
    return ans
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}

File: 0000-0999/0011.ContainerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        ans = 0
        left, right = 0, len(height)-1
        '''
        while left < right:
            ans = max(ans, min(height[left], height[right])*(right-left))
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        '''
        while left < right:
            h = min(height[left], height[right])
            ans = max(ans, h*(right-left))
            while left < right and height[left] <= h:
                left += 1
            while left < right and height[right] <= h:
                right -= 1
        return ans
        